File: cashier-conundrum/solutions/sol.py
x, y = map(int, input().split())
denominations = [int(d) for d in input().split()]
assert len(denominations) == x

for _ in range(y):
    coins = 0
    change = int(input())
    output = []
    for value in denominations:
        num_coin = change // value
        coins += num_coin
        change = change % value
        output.append(num_coin)
    print(*output)

File: cashier-conundrum/solutions/sol.java
import java.util.*;
import java.lang.*;
import java.io.*;

public class Solution {
    public static void main (String[] args) {
        Scanner sc = new Scanner(System.in);
        int x = sc.nextInt();
        int y = sc.nextInt();
        int[] denominations = new int[x];
        for(int i=0; i<x; i++) {
            denominations[i] = sc.nextInt();
        }
        for(int i=0; i<y; i++) {
            int coins = 0;
            int change = sc.nextInt();
            int[] output = new int[x];
            for(int j=0; j<x; j++) {
                int num_coin = change / denominations[j];
                coins += num_coin;
                change = change % denominations[j];
                output[j] = num_coin;
            }
            for(int j=0; j<x; j++) {
                System.out.print(output[j] + " ");
            }
            System.out.println();
        }
    }
}